src/Request.ts

```diff
@@ -0,0 +1,44 @@
+import Participant from './db/models/participant';
+import { InstanceOfModel } from './db/util/types';
+import * as hapi from '@hapi/hapi';
+import * as restify from 'restify';
+
+interface ExpandedRequest {
```

Comment:
Hmm this class also feels like something that's a little too specific to the way that we expect requests to be constructed for restify using our middlewear in `hpc_service` only, and not something that feels generally applicable for the legacy and new API.

The general library should not need to be aware of specific implementations / libraries that we're using to implement the rest API servers (and therefore should not need to have any restify / hapi types included at all). Instead, any interfaces should only really include stuff that's common across both usages.